looper/utils.py

```diff
@@ -0,0 +1,157 @@
+""" Helpers without an obvious logical home. """
+
+from collections import defaultdict
+import logging
+import os
+import yaml
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+
+class CommandChecker(object):
```

Comment:
What's the motivation to nest it within another class? It seems like it could have use in isolation. The only reason that this is really a class is for several data structures to store results of the checks. I could see a case being made for a function version that just creates and instance and returns the `failed` result, since I think that'd be the most common use. The other data structures are just to provide more fine-grained inspection capability.